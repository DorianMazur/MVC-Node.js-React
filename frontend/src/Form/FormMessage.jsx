import React from 'react';
import PropTypes from 'prop-types';
import { Message } from 'semantic-ui-react';

const FormView = props => {
  if (!props.show) return null;
  return (
    <Message success={!props.error} error={props.error}>
      {props.message || 'Unexpected error ocurred!'}
    </Message>
  );
};

FormView.propTypes = {
  show: PropTypes.bool,
  message: PropTypes.string,
  error: PropTypes.bool,
};

export default FormView;
