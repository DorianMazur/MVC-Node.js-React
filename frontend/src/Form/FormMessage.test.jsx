import '@testing-library/jest-dom';
import React from 'react';

import { render } from '@testing-library/react';
import FormMessage from './FormMessage.jsx';

test('FormMessage render correctly (error message)', () => {
  const container = render(<FormMessage show={true} message={'Error Test Message!'} error={true} />);
  expect(container.baseElement).toMatchSnapshot();
});

test('FormMessage render correctly (success message)', () => {
  const container = render(<FormMessage show={true} message={'Success Test Message!'} />);
  expect(container.baseElement).toMatchSnapshot();
});

test('FormMessage render correctly (no message)', () => {
  const container = render(<FormMessage show={false} message={'Error Test Message!'} erorr={true} />);
  expect(container.baseElement).toMatchSnapshot();
});
