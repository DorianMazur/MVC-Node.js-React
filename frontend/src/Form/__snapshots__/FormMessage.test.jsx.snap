// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormMessage render correctly (error message) 1`] = `
<body>
  <div>
    <div
      class="ui error message"
    >
      Error Test Message!
    </div>
  </div>
</body>
`;

exports[`FormMessage render correctly (no message) 1`] = `
<body>
  <div />
</body>
`;

exports[`FormMessage render correctly (success message) 1`] = `
<body>
  <div>
    <div
      class="ui success message"
    >
      Success Test Message!
    </div>
  </div>
</body>
`;
